spec/unit/prepare.spec.js

```diff
@@ -0,0 +1,935 @@
+        );
+
+        done();
+    });
+
+    it('Test#003 : Should detech incorrect configrations (missing foreground) for adaptive icon and throw an error.', function (done) {
+        const updateIcons = prepare.__get__('updateIcons');
+
+        // mock data.
+        cordovaProject.projectConfig.getIcons = function () {
+            return [{
+                src: undefined,
+                target: undefined,
+                density: 'mdpi',
+                platform: 'android',
+                width: undefined,
+                height: undefined,
+                background: 'res/icon/android/mdpi-background.png',
+                foreground: undefined
+            }];
+        };
+
+        expect(function () {
+            updateIcons(cordovaProject, platformResourcesDir);
+        }).toThrow(
+            new CordovaError('One of the following attributes are set but missing the other for the density type: mdpi. Please ensure that all require attributes are defined.')
+        );
+
+        done();
+    });
+
+    it('Test#004 : Should detech incorrect configrations (missing background) for adaptive icon and throw an error.', function (done) {
+        const updateIcons = prepare.__get__('updateIcons');
+
+        // mock data.
+        cordovaProject.projectConfig.getIcons = function () {
+            return [{
+                src: undefined,
+                target: undefined,
+                density: 'mdpi',
+                platform: 'android',
+                width: undefined,
+                height: undefined,
+                background: undefined,
+                foreground: 'res/icon/android/mdpi-foreground.png'
+            }];
+        };
+
+        expect(function () {
+            updateIcons(cordovaProject, platformResourcesDir);
+        }).toThrow(
+            new CordovaError('One of the following attributes are set but missing the other for the density type: mdpi. Please ensure that all require attributes are defined.')
+        );
+
+        done();
+    });
+
+    it('Test#005 : Should detech incorrect configrations and throw an error.', function (done) {
+        const updateIcons = prepare.__get__('updateIcons');
+
+        // mock data.
+        cordovaProject.projectConfig.getIcons = function () {
+            return [{
+                src: undefined,
+                target: undefined,
+                density: 'mdpi',
+                platform: 'android',
+                width: undefined,
+                height: undefined,
+                background: undefined,
+                foreground: undefined
+            }];
+        };
+
+        expect(function () {
+            updateIcons(cordovaProject, platformResourcesDir);
+        }).toThrow(
+            new CordovaError('One of the following attributes are set but missing the other for the density type: mdpi. Please ensure that all require attributes are defined.')
+        );
+
+        done();
```

Comment:
Are these tests asynchronous? If not, we don't need this. If so, I suggest returning promises to Jasmine instead of using the done callback.